src/server/server_impl.go

```diff
@@ -79,24 +78,28 @@ func (server *server) AddDebugHttpEndpoint(path string, help string, handler htt
 // example usage from cURL with domain "dummy" and descriptor "perday":
 // echo '{"domain": "dummy", "descriptors": [{"entries": [{"key": "perday"}]}]}' | curl -vvvXPOST --data @/dev/stdin localhost:8080/json
 func NewJsonHandler(svc pb.RateLimitServiceServer) func(http.ResponseWriter, *http.Request) {
-	// Default options include enums as strings and no identation.
-	m := &jsonpb.Marshaler{}
-
 	return func(writer http.ResponseWriter, request *http.Request) {
 		var req pb.RateLimitRequest
 
 		ctx := context.Background()
 
-		if err := jsonpb.Unmarshal(request.Body, &req); err != nil {
+		body, err := io.ReadAll(request.Body)
+		if err != nil {
+			logger.Warnf("error: %s", err.Error())
+			writeHttpStatus(writer, http.StatusBadRequest)
+			return
+		}
+
+		if err := protojson.Unmarshal(body, &req); err != nil {
 			logger.Warnf("error: %s", err.Error())
-			http.Error(writer, err.Error(), http.StatusBadRequest)
```

Comment:
Any particular reason we're removing the more descriptive error msgs in the responses? Is it for consistency, security?

Reply:
Ah my apologies! I have updated the PR description to provide more clarity around why I did this.

In this exact line you commented on we are still doing the same thing; I just refactored it into a shared function for consistency.